core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -820,6 +820,7 @@ private[spark] class TaskSetManager(
         s"on ${info.host} (executor ${info.executorId}) ($tasksSuccessful/$numTasks)")
       // Mark successful and stop if all the tasks have succeeded.
       successful(index) = true
+      numFailures(index) = 0
```

Comment:
Our current definition of `spark.task.maxFailures` is slightly ambiguous in this regard.
From docs:
> 
>     Number of failures of any particular task before giving up on the job.
>     The total number of failures spread across different tasks will not cause the job
>     to fail; a particular task has to fail this number of attempts.
>     Should be greater than or equal to 1. Number of allowed retries = this value - 1.
> 

I would argue any successful task should reset the counter, as that was the original intent; and given that - this change (to reset to 0) should be fine.

Reply:
Shall we update the doc comment for spark.task.maxFailures ? Sorry about that maybe I don't understand your intention very clearly， @mridulm

Reply:
Apologies for the confusion @wangshengjie123 !
I was essentially detailing my reasoning behind why I am fine with the numFailures reset - not proposing further changes :-)

Reply:
> 
My bad, I had the doubts mentioned above before and want to update the doc, so i got more confused. Thanks for explain again, @mridulm

Reply:
> Our current definition of spark.task.maxFailures is slightly ambiguous in this regard.

So shall we update the doc? @mridulm